c++/src/Reader.hh

```diff
@@ -88,10 +88,12 @@ namespace orc {
     // Select a field by id
     void updateSelectedByFieldId(std::vector<bool>& selectedColumns, uint64_t fieldId);
     // Select a type by id
-    void updateSelectedByTypeId(std::vector<bool>& selectedColumns, uint64_t typeId);
+    void updateSelectedByTypeId(std::vector<bool>& selectedColumns, uint64_t typeId,
+                                const RowReaderOptions::TypeReadIntents* readIntents = nullptr);
```

Comment:
Shall we use reference instead of pointer?

Reply:
Done.